<?php

uses(RefreshDatabase::class);

test('resend returns an ok response', function (): void {
    $this->markTestIncomplete('This test case was generated by Shift. When you are ready, remove this line and complete this test case.');

    $user = \App\Models\User::factory()->create();

    $response = $this->actingAs($user)->post(route('resend'), [
        // TODO: send request data
    ]);

    $response->assertOk();

    // TODO: perform additional assertions
});

test('resend aborts with a 404', function (): void {
    $this->markTestIncomplete('This test case was generated by Shift. When you are ready, remove this line and complete this test case.');

    $user = \App\Models\User::factory()->create();

    // TODO: perform additional setup to trigger `abort_if(404)`...

    $response = $this->actingAs($user)->post(route('resend'), [
        // TODO: send request data
    ]);

    $response->assertNotFound();
});

test('show verification returns an ok response', function (): void {
    $this->markTestIncomplete('This test case was generated by Shift. When you are ready, remove this line and complete this test case.');

    $user = \App\Models\User::factory()->create();

    $response = $this->actingAs($user)->get(route('verificationNeeded'));

    $response->assertOk();
    $response->assertViewIs('auth.twostep-exceeded');

    // TODO: perform additional assertions
});

test('show verification aborts with a 404', function (): void {
    $this->markTestIncomplete('This test case was generated by Shift. When you are ready, remove this line and complete this test case.');

    $user = \App\Models\User::factory()->create();

    // TODO: perform additional setup to trigger `abort_if(404)`...

    $response = $this->actingAs($user)->get(route('verificationNeeded'));

    $response->assertNotFound();
});

test('verify returns an ok response', function (): void {
    $this->markTestIncomplete('This test case was generated by Shift. When you are ready, remove this line and complete this test case.');

    $user = \App\Models\User::factory()->create();

    $response = $this->actingAs($user)->post(route('verify'), [
        // TODO: send request data
    ]);

    $response->assertOk();

    // TODO: perform additional assertions
});

test('verify aborts with a 404', function (): void {
    $this->markTestIncomplete('This test case was generated by Shift. When you are ready, remove this line and complete this test case.');

    $user = \App\Models\User::factory()->create();

    // TODO: perform additional setup to trigger `abort_if(404)`...

    $response = $this->actingAs($user)->post(route('verify'), [
        // TODO: send request data
    ]);

    $response->assertNotFound();
});

test('verify aborts with a 404', function (): void {
    $this->markTestIncomplete('This test case was generated by Shift. When you are ready, remove this line and complete this test case.');

    $user = \App\Models\User::factory()->create();

    // TODO: perform additional setup to trigger `abort(404)`...

    $response = $this->actingAs($user)->post(route('verify'), [
        // TODO: send request data
    ]);

    $response->assertNotFound();
});

// test cases...
